### ISSUE TITLE
Changes in shared package from monorepo are not reflected in *.vue files

### ISSUE THREAD

**Issue by alexvoedi** [vuejs/language-tools#3392]
I am working on a monorepo with a shared folder (see https://github.com/alexvoedi/vite-nest-monorepo). Whenever I change a type in the shared folder I want the changes to be reflected in the vue web app. 

Example:

I change the type of `id` in the shared folder:

![image](https://github.com/vuejs/language-tools/assets/37340862/60bc8d88-a533-4657-9436-1c880dfdc1e8)

After building the shared package, the changes should be reflected in the vue web app. This works in `*.ts` files

![image](https://github.com/vuejs/language-tools/assets/37340862/1d8057d8-8a45-4289-abbe-0392b5ca13a4)

but not in `*.vue` files

![image](https://github.com/vuejs/language-tools/assets/37340862/f9516446-f3e8-43d2-8318-b3ffc4cd00fa)

The problem must be related to volar because when I enable takeover mode the problem is not any more limited to `*.vue` files but also `*.ts` files. Disabling takeover mode limits this problem to only `*.vue` files.

Note that this is a pnpm workspace and the monorepo is powered by Turborepo (even though I am sure the problem has nothing to do with this setup).

**Comment by rchl** [comment]
Probably a dup of https://github.com/vuejs/language-tools/issues/3229

**Comment by jmroon** [comment]
I've found that Volar has tried to do performance optimizations that have resulted in the breaking of typescript in a monorepo. In a tsconfig paths based monorepo (i.e. integrated monorepo), I've found that I need to specifically add monorepo libraries to the includes section of my app's package.json.

For pnpm workspaces, I think you'll have to add the symlinked node_modules folder to your tsconfig's includes property. i.e. `include = [ "node_modules/my-monorepo-library/**.*.ts", "node_modules/my-monorepo-library/**.*.vue" ]
